### PR TITLE
[IOTDB-6175] Add TopKNode for `order by time|expression limit N align by device` query and push down LIMIT value.

### PR DESCRIPTION
**Pull request by Beyyes**
## Description

see design doc in: https://apache-iotdb.feishu.cn/docx/AQHwdtenhopyAQxTN2hcnXVwnoh

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanVisitor.java [hunk 1]
```diff
@@ -364,6 +367,17 @@ public PlanNode visitQueryBody(
     return planBuilder.getRoot();
   }
 
+  private long pushDownLimitToScanNode(QueryStatement queryStatement) {
```
[the remark this answers is not in the document]
Comment (Beyyes) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanVisitor.java:370]: s1:  1,2,3
s2: 4,5,6
s3: 7,8,9  offset 1 limit 1

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -759,22 +774,48 @@ public LogicalPlanBuilder planDeviceView(
...
-          new MergeSortNode(
-              context.getQueryId().genPlanNodeId(), orderByParameter, outputColumnNames);
-      for (Map.Entry<String, PlanNode> entry : deviceNameToSourceNodesMap.entrySet()) {
-        String deviceName = entry.getKey();
-        PlanNode subPlan = entry.getValue();
-        SingleDeviceViewNode singleDeviceViewNode =
-            new SingleDeviceViewNode(
+      // order by time, device + limit can be optimized to TopK implementation
+      // in the future, offset can also be pushed down to ScanNode and use TopK process
+      if (queryStatement.hasLimit()
```
Comment (Beyyes) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanBuilder.java:779]: optimize this judgement

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanBuilder.java [hunk 3]
```diff
@@ -192,27 +196,37 @@ public LogicalPlanBuilder planRawDataSource(
...
             new SeriesScanNode(
-                context.getQueryId().genPlanNodeId(), (MeasurementPath) path, scanOrder);
-        seriesScanNode.setTimeFilter(timeFilter);
-        // TODO: push down value filter
-        seriesScanNode.setValueFilter(timeFilter);
+                context.getQueryId().genPlanNodeId(),
+                (MeasurementPath) path,
+                scanOrder,
+                timeFilter,
+                timeFilter,
```
Comment (wangchao316) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanBuilder.java:208]: twice timeFilter paremeter?

Comment (Beyyes) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanBuilder.java:209]: yes, value filter can not be pushed down now, and we will implement it in few months.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**